src/Widgets/MultitaskingView.vala

```diff
@@ -178,32 +178,34 @@ namespace Gala {
                 return true;
             }
 
-            if (scroll_event.direction != ScrollDirection.SMOOTH) {
+            if (scroll_event.direction == ScrollDirection.SMOOTH ||
+                scroll_event.scroll_source == ScrollSource.FINGER ||
+                scroll_event.get_source_device ().get_device_type () == Clutter.InputDeviceType.TOUCHPAD_DEVICE) {
                 return false;
             }
 
-            double dx, dy;
-#if VALA_0_32
-            scroll_event.get_scroll_delta (out dx, out dy);
-#else
-            var event = (Event*)(&scroll_event);
-            event->get_scroll_delta (out dx, out dy);
-#endif
-
-            // concept from maya to detect mouse wheel
-            if (Math.fabs (dy) == 1.0) {
-                var direction = dy > 0 ? MotionDirection.RIGHT : MotionDirection.LEFT;
+            Meta.MotionDirection direction;
+            switch (scroll_event.direction) {
+                case ScrollDirection.UP:
+                case ScrollDirection.LEFT:
+                    direction = MotionDirection.LEFT;
+                    break;
+                case ScrollDirection.DOWN:
+                case ScrollDirection.RIGHT:
+                default:
+                    direction = MotionDirection.RIGHT;
+                    break;
+            }
 
-                unowned Meta.WorkspaceManager manager = display.get_workspace_manager ();
-                var active_workspace = manager.get_active_workspace ();
-                var new_workspace = active_workspace.get_neighbor (direction);
+            unowned Meta.WorkspaceManager manager = display.get_workspace_manager ();
+            var active_workspace = manager.get_active_workspace ();
+            var new_workspace = active_workspace.get_neighbor (direction);
 
-                if (active_workspace != new_workspace) {
-                    new_workspace.activate (display.get_current_time ());
-                }
+            if (active_workspace != new_workspace) {
+                new_workspace.activate (display.get_current_time ());
             }
 
-            return false;
+            return true;
```

Comment:
As long as we're changing this line, can we use the Gdk macro so the intent is more clear?

```suggestion
            return EVENT_STOP;
```

Reply:
Eh, since this isn't a GDK event handler, I'd be tempted not to. Yes, it's almost an impossibility that will ever be anything other than `true` but since the two things aren't related, it doesn't seem right to me.

We also need to be using less GTK/GDK in Gala, because it causes issues with Wayland. Again, yes, this is just a constant so it wouldn't be part of the problem. But if we can remove GTK altogether, we can drop the dependency.

Reply:
There's `CLUTTER_EVENT_STOP`, though doesn't seem like it's in the vapi.

Reply:
@Exalm is right, GNOME Shell uses `Clutter.EVENT_PROPAGATE/EVENT_STOP`. I tried to do the same, but as he mentioned, they are not in the vapi.

I guess it's something we can address in a different PR.